test/e2e/testsuites/pre_provisioned_keyvault_tester.go

```diff
@@ -0,0 +1,372 @@
+	"sigs.k8s.io/blob-csi-driver/test/e2e/driver"
+	"sigs.k8s.io/blob-csi-driver/test/utils/credentials"
+)
+
+var (
+	subscriptionID    string
+	resourceGroupName string
+	location          string
+	vaultName         string
+	TenantID          string
+	cloud             string
+	clientID          string
+	clientSecret      string
+)
+
+// PreProvisionedKeyVaultTest will provision required PV(s), PVC(s) and Pod(s)
+// Testing that the Pod(s) can be created successfully with provided Key Vault
+// which is used to store storage account name and key(or sastoken)
+type PreProvisionedKeyVaultTest struct {
+	CSIDriver driver.PreProvisionedVolumeTestDriver
+	Pods      []PodDetails
+	Driver    *blob.Driver
+}
+
+func (t *PreProvisionedKeyVaultTest) Run(client clientset.Interface, namespace *v1.Namespace) {
+	e2eCred, err := credentials.ParseAzureCredentialFile()
+	framework.ExpectNoError(err, fmt.Sprintf("Error ParseAzureCredentialFile: %v", err))
+
+	subscriptionID = e2eCred.SubscriptionID
+	resourceGroupName = e2eCred.ResourceGroup
+	location = e2eCred.Location
+	TenantID = e2eCred.TenantID
+	cloud = e2eCred.Cloud
+	clientID = e2eCred.AADClientID
+	clientSecret = e2eCred.AADClientSecret
+	vaultName = names.SimpleNameGenerator.GenerateName("blob-csi-test-kv-")
+
+	for _, pod := range t.Pods {
+		for n, volume := range pod.Volumes {
+			// In the method GetStorageAccountAndContainer, we can get an account key of the blob volume
+			// by calling azure API, but not the sas token...
+			accountName, accountKey, _, containerName, err := t.Driver.GetStorageAccountAndContainer(context.TODO(), volume.VolumeID, nil, nil)
+			framework.ExpectNoError(err, fmt.Sprintf("Error GetStorageAccountAndContainer from volumeID(%s): %v", volume.VolumeID, err))
+
+			azureCred, err := azidentity.NewDefaultAzureCredential(nil)
+			framework.ExpectNoError(err)
+
+			ginkgo.By("creating KeyVault...")
+			vault, err := createVault(context.TODO(), azureCred)
+			framework.ExpectNoError(err)
+			defer cleanVault(context.TODO(), azureCred)
+
+			ginkgo.By("creating secret for storage account key...")
+			accountKeySecret, err := createSecret(context.TODO(), azureCred, accountName+"-key", accountKey)
+			framework.ExpectNoError(err)
+
+			pod.Volumes[n].ContainerName = containerName
+			pod.Volumes[n].StorageAccountname = accountName
+			pod.Volumes[n].KeyVaultURL = *vault.Properties.VaultURI
+			pod.Volumes[n].KeyVaultSecretName = *accountKeySecret.Name
+			// test for Account key
+			ginkgo.By("test storage account key...")
+			run(pod, client, namespace, t.CSIDriver)
+
+			ginkgo.By("generate SAS token...")
+			sasToken := generateSASToken(accountName, accountKey)
+
+			ginkgo.By("creating secret for SAS token...")
+			accountSASSecret, err := createSecret(context.TODO(), azureCred, accountName+"-sas", sasToken)
+			framework.ExpectNoError(err)
+
+			pod.Volumes[n].KeyVaultSecretName = *accountSASSecret.Name
+			// TODO: test for SAS token
+			// ginkgo.By("test SAS token...")
```

Comment:
sas token e2e test is actually different case, pls create another github issue to track, thanks.